msal/src/main/java/com/microsoft/identity/client/NativeAuthPublicClientApplication.kt

```diff
@@ -524,6 +536,402 @@ class NativeAuthPublicClientApplication(
+                            correlationId = result.correlationId
+                        ),
+                        invalidAttributes = result.invalidAttributes
+                    )
+                }
+
+                is SignUpCommandResult.PasswordRequired -> {
+                    SignUpResult.PasswordRequired(
+                        nextState = SignUpPasswordRequiredState(
+                            flowToken = result.signupToken,
+                            username = username,
+                            config = nativeAuthConfig
+                        )
+                    )
+                }
+
+                is SignUpCommandResult.Complete -> {
+                    SignUpResult.Complete(
+                        nextState = SignInAfterSignUpState(
+                            signInVerificationCode = result.signInSLT,
+                            username = username,
+                            config = nativeAuthConfig
+                        )
+                    )
+                }
+
+                is INativeAuthCommandResult.Redirect -> {
+                    SignUpResult.BrowserRequired(
+                        error = BrowserRequiredError(
+                            correlationId = result.correlationId
+                        )
+                    )
+                }
+
+                is INativeAuthCommandResult.UnknownError -> {
+                    SignUpResult.UnexpectedError(
+                        error = GeneralError(
+                            errorMessage = result.errorDescription,
+                            error = result.error,
+                            correlationId = result.correlationId,
+                            details = result.details,
+                            exception = result.exception
+                        )
+                    )
+                }
+
+                is SignUpCommandResult.InvalidPassword -> {
+                    Logger.warn(
+                        TAG,
+                        "Unexpected result $result"
+                    )
+                    SignUpResult.UnexpectedError(
+                        error = GeneralError(
+                            errorMessage = "Unexpected state",
+                            error = "unexpected_state",
+                            correlationId = result.correlationId
+                        )
+                    )
+                }
+
+                is SignUpCommandResult.AuthNotSupported -> {
+                    Logger.warn(
+                        TAG,
+                        "Unexpected result $result"
+                    )
+                    SignUpResult.UnexpectedError(
+                        error = GeneralError(
+                            errorMessage = "Unexpected state",
+                            error = "unexpected_state",
+                            correlationId = result.correlationId
+                        )
+                    )
+                }
```

Comment:
is there anything different in how signup and signupwithpassword handle the command results? Feel like most if not all of this can be pulled out into a shared helper method.

Reply:
I tried to create it but it cannot be done easily. The return types for these methods are different classes `SignUpResult` and `SignUpUsingPasswordResult`. The `SignUpUsingPasswordResult` does not derive from `SignUpResult` so I cannot find a common type that will be suitable for the helper method.

Reply:
I looked into today to see if we can change the hierarchies of `SignUpResult ` and `SignUpUsingPasswordResult` classes to have a common method. We cannot do it without making wholesale changes to class hierarchies.